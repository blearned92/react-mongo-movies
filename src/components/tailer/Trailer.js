import ReactPlayer from 'react-player';

const Trailer = ({trailer}) => {

        const key = trailer.key;
        return(
            <ReactPlayer  
                controls={true} 
                playing={false}
                url={`https://www.youtube.com/watch?v=${key}`}
                light={`https://img.youtube.com/vi/${key}/0.jpg`}
                width='100%' 
                height='100%'
                /> 
    )
}

export default Trailer;