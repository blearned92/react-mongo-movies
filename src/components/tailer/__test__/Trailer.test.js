import { render, fireEvent, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Trailer from "../Trailer";

describe("Trailer",()=>{
    const trailerObject = {key:"640146"};
    // it('should render',()=>{
    //     //render the component
    //     render(<Trailer trailer={trailerObject}/>)
    //     const trailer = screen.findByTestId("trailer-component");
    //     expect(trailer).toBe;
    // })
    it('should match DOM snapshot',()=>{
        const tree = renderer.create(<Trailer trailer={trailerObject}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})
