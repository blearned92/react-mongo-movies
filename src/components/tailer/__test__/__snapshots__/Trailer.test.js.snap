// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trailer should match DOM snapshot 1`] = `
<div
  data-testid="trailer-component"
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
/>
`;
